ValveResourceFormat/Resource/ResourceTypes/ModelAnimation/Animation.cs

```diff
@@ -152,19 +152,15 @@ private static IKeyValueCollection GetAnimationData(Resource resource)
         public Matrix4x4[] GetAnimationMatrices(AnimationFrameCache frameCache, float time, Skeleton skeleton)
         {
             // Create output array
-            var matrices = new Matrix4x4[skeleton.AnimationTextureSize];
-
-            for (var i = 0; i < skeleton.AnimationTextureSize; i++)
-            {
-                matrices[i] = Matrix4x4.Identity;
-            }
+            var matrices = Enumerable.Range(0, skeleton.Bones.Length)
```

Comment:
old loop was fine.

Reply:
I've been changing code a lot, right now it doesn't even require filling identity matrices.

Changed to plain `new Matrix4x4[skeleton.Bones.Length]`.